datafusion/expr/src/expr_fn.rs

```diff
@@ -704,6 +704,8 @@ scalar_expr!(
     array delimiter,
     "converts each element to its text representation."
 );
+nary_scalar_expr!(ArrayUnion, array_union, "returns an array of the elements in the union of array1 and array2 without duplicates.");
```

Comment:
should we use scalar_expr?

Reply:
Fixed with https://github.com/apache/arrow-datafusion/commit/648ae5d659142082d085418dc7b84ad857ba8136